src/app/modules/app/pages/classrooms/components/classroom-form/classroom-form.component.ts

```diff
@@ -0,0 +1,194 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import {
+  FormArray,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { MAT_FORM_FIELD_DEFAULT_OPTIONS } from '@angular/material/form-field';
+import { map } from 'rxjs';
+import { weekdayOptions } from 'src/app/shared/constants/weekday-options';
+import { SelectOption } from 'src/app/shared/interfaces/select-options';
+import { Classroom } from 'src/app/shared/models/Classroom';
+import { ClassroomService } from 'src/app/shared/services/classroom.service';
+import { CourseService } from 'src/app/shared/services/course.service';
+import { ProgramService } from 'src/app/shared/services/program.service';
+import { transform24TimeTo12Time } from 'src/app/shared/utils/date-mapper';
+
+export type ScheduleForm = FormGroup<{
+  weekday: FormControl<string | null>;
+  startingTime: FormControl<string | null>;
+  endingTime: FormControl<string | null>;
+}>;
+
+@Component({
+  selector: 'app-classroom-form',
+  templateUrl: './classroom-form.component.html',
+  styleUrls: ['./classroom-form.component.sass'],
+  providers: [
+    {
+      provide: MAT_FORM_FIELD_DEFAULT_OPTIONS,
+      useValue: { appearance: 'outline', floatLabel: 'always' },
+    },
+  ],
+})
+export class ClassroomFormComponent implements OnInit {
+  title!: string;
+  classroomForm!: FormGroup;
+  programOptions: SelectOption[] = [];
+  courseOptions: SelectOption[] = [];
+  weekdayOptions = weekdayOptions;
+
+  constructor(
+    public readonly classroomService: ClassroomService,
+    public readonly programService: ProgramService,
+    public readonly courseService: CourseService,
+    private readonly formBuilder: FormBuilder,
+    private readonly dialogRef: MatDialogRef<ClassroomFormComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: { item: Classroom }
+  ) {}
+
+  private initializeForm(): void {
+    this.classroomForm = this.formBuilder.group({
+      course: [
+        this.data ? this.data.item.courseObject?.id?.toString() : '',
+        [Validators.required],
+      ],
+      program: [
+        this.data ? this.data.item.programObject?.id.toString() : '',
+        [Validators.required],
+      ],
+      startingDate: [
+        this.data ? this.data.item.starts : '',
+        [Validators.required],
+      ],
+      schedules: new FormArray<ScheduleForm>([]),
+    });
+  }
+
+  get schedules(): FormArray {
+    return this.classroomForm.get('schedules') as FormArray<ScheduleForm>;
+  }
+
+  addSchedule() {
+    const schedule: ScheduleForm = this.formBuilder.group({
+      weekday: ['', [Validators.required]],
+      startingTime: ['', [Validators.required]],
+      endingTime: ['', [Validators.required]],
```

Comment:
Add validator for ending time can't be prior to starting time and viceversa
![image](https://github.com/JU-DEV-LatamBootcamp/Klockan-Frontend/assets/151551082/12d941e6-c71f-4bf7-aae3-a85365cebe3d)